tests/kernel/interrupt/testcase.yaml

```diff
@@ -6,3 +6,12 @@ tests:
       - kernel
       - interrupt
     filter: not CONFIG_TRUSTED_EXECUTION_NONSECURE
+  arch.shared_interrupt:
+    arch_allow:
+      - arm64
+      - arm
```

Comment:
I really would like to see at RISCV in this list.

Reply:
As far as I can tell, the RISCV-based QEMU SoCs use multi-level interrupts which is currently not supported (should be added later on alongside the architectures that use it). Ideally, the support for multi-level interrupts should be added by someone a bit more familiar with it. Is this a deal-breaker?

Reply:
In that case this should be `depends on !CONFIG_MULTI_LEVEL_INTERRUPTS` but even with that you can still definitely use for example the interrupt ID 3 (software interrupt) to test this on RISCV.

Reply:
Added `depends on !MULTI_LEVEL_INTERRUPTS` to the `SHARED_INTERRUPTS` configuration. Allowing the test to run on RISCV-based platform would mean having to remove `depends on !MULTI_LEVEL_INTERRUPTS` and adding `ARCH_HAS_SHARED_IRQ_SUPPORT` to RISCV which would mean the shared interrupts should work for all interrupt lines. This is wrong. I guess the best I can do is test this locally?

Reply:
yeah but I mean, in 90% of the RISCV platform you will have a multi level interrupt management because PLIC & company require that so I'm ready to bet that this will be never implemented for RISCV then because no one will bother with it so this will be never out of experimental and actually used.

This is why I don't like half-assessed features that work only on a very restricted selection of architectures.

Can you explain the technical difficulties in making this work also with multi level interrupts?

Reply:
Understood. Since this is the case I'll also make it work with multi-level interrupts.


> Can you explain the technical difficulties in making this work also with multi level interrupts?

Mainly my lack of knowledge on the matter. Otherwise I expect it to be fairly easy as it's similar to how it was done for `_sw_isr_table`.